### PR TITLE
Check that the DB version is set before deploying

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: commands/platformsh_hooks.go [hunk 1]
```diff
@@ -1,13 +1,94 @@
...
 
 var platformshBeforeHooks = map[string]console.BeforeFunc{
+	"cloud:environment:push": func(c *console.Context) error {
+		// check that project has a DB and that server version is set properly
+		projectDir, err := getProjectDir(c.String("dir"))
+		if err != nil {
+			return err
+		}
+		if len(platformsh.FindLocalApplications(projectDir)) > 1 {
+			// not implemented yet as more complex
```
[the remark this answers is not in the document]
Comment (fabpot) [reply commands/platformsh_hooks.go:39]: I want the less friction possible. So, if I'm not sure about how to do the check, I want it to be silent.

[the remark this answers is not in the document]
Comment (tucksaun) [reply commands/platformsh_hooks.go:39]: 👍

File: commands/db_versions.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		if !strings.Contains(line, "serverVersion=") {
+			return "", nil
+		}
+		return strings.TrimRight(strings.Split(strings.Split(line, "serverVersion=")[1], "&")[0], "\""), nil
+	}
+	return "", nil
+}
+
+func readDBVersionFromDoctrineConfigYAML(projectDir string) (string, error) {
+	path := filepath.Join(projectDir, "config", "packages", "doctrine.yaml")
```
[the remark this answers is not in the document]
Comment (fabpot) [reply commands/db_versions.go:69]: But then, you need to deal with multiple connections and what not, I'm trying to only support the happy path :)

File: commands/platformsh_hooks.go [hunk 3]
```diff
@@ -1,13 +1,94 @@
...
+			return nil
+		}
+
+		dbName, dbVersion, err := readDBVersionFromPlatformServiceYAML(projectDir)
+		if err != nil {
+			return nil
```
[the remark this answers is not in the document]
Comment (fabpot) [reply commands/platformsh_hooks.go:45]: Yep, on purpose. I want to be silent most of the time. If this file is broken, there is no way you can even develop locally, so it won't happen ;)

**Pull request merged**